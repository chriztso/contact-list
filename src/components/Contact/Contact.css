.contact{
    display: flex;
    border: 1px solid rgb(233, 234, 240);
    align-items: center;
    height: 44px;
    position: relative;
}

.contact:hover{
    background-color: rgb(250,251,255);
}

.contact__checkbox{
    margin-left: 12px;
    margin-right: 12px;
}

.contact__name{
    display: flex;
    width: 20%; 
    align-items: center;
    color: rgb(68, 107, 222);
    font-weight: 600;
}

.contact__initials{
    height: 25px;
    width: 25px;
    background-color: rgb(232,224,252);
    color: black;
    font-weight: bold;
    border-radius: 50%;
    display: flex;
    align-items: center;
    margin-right: 5px;
    justify-content: center;
    font-size: 12px;
} 

.contact__value{
    width: 15%; 
}

.contact__location{
    width: 15%; 
}

.contact__deals{
    width: 10%; 
}

.contact__tags{
    width: 26%; 
}

.contact__email-button{
    position: absolute;
    right: 12px;
    display: flex;
    border: 1px solid rgb(207, 211, 223);
    border-radius: 5px;
    align-items: center;
    height: 30px;
}

.contact__email{
    border-right: 1px solid rgb(207, 211, 223);
    display: flex; 
    align-items: center;
    height: 100%;
    padding: 0px 10px;
    font-weight: 500;
}

.contact__down-arrow{
    display: flex; 
}