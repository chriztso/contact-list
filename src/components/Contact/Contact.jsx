import React, { useState } from 'react'
import ArrowDropDownIcon from '@material-ui/icons/ArrowDropDown';

import './Contact.css'

const Contact = (props) => {
    const {
        name,
        initials,
        contactTagsStr,
        numberOfDeals,
        dealsTotalValue,
        geoAddressStr
    } = props

    const [showButton, setShowButton] = useState(false)

    return (
        <div className="contact" 
          onMouseEnter={() => {setShowButton(true)}}
          onMouseLeave={() => {setShowButton(false)}}
        >
            <input type="checkbox" className="contact__checkbox"></input>
            <div className="contact__name">
                <div className="contact__initials">
                    {initials}
                </div>
                <div>{name}</div>
            </div>
            <span className="contact__value">${dealsTotalValue}</span>
            <span className="contact__location">{geoAddressStr}</span>
            <span className="contact__deals">{numberOfDeals}</span>
            <span className="contact__tags">{contactTagsStr}</span>
            {showButton && <div className="contact__email-button">
              <div className="contact__email">Email</div>
              <div className="contact__down-arrow"><ArrowDropDownIcon/></div>
            </div>}
        </div>
    )
}

export default Contact