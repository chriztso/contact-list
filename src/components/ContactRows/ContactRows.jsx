import Contact from '../Contact/Contact'
import { getStateAbbreviation } from '../../utils/format'   
import convertToUSD from '../../utils/convert'

const ContactRows = ( props ) => {
    const {
        contacts,
        dealsObject,
        tagsObject,
        contactTagsObject,
        geoIpsObject,
        geoAddressesObject
    } = props

    const getContactTags = (contactTags) => {
        let contactTagsArr = []
        //go through each contactTag ID in contact property array, 
        contactTags.forEach(contactTagID => {
            //access the object in contactTagObjects with contact tag ID
            const contactTagObject = contactTagsObject[contactTagID]
            //get tag ID from contactTag object 
            const tagId = contactTagObject["tag"]
            //access the object in tagObjects
            const tagObject = tagsObject[tagId]
            //get tag from tag object
            const tag = tagObject["tag"]
            //push to array
            contactTagsArr.push(tag)
        })
        console.log('CT ARR', contactTagsArr)
        //create string from contact tags array
        const contactTagsStr = contactTagsArr.join(', ')
        console.log("CT STR", contactTagsStr)
        return contactTagsStr
    }

    const getDealsTotalValue = (deals) => {
        let dealsTotalValueUSD = 0
        //go through each deal ID in contact property array
        deals.forEach(dealID => {
            //access the deal object in dealObjects with each deal ID
            const dealObj = dealsObject[dealID]
            //get value and currency of deal object and add to dealsTotalValue
            const { value, currency } = dealObj
            //convert to USD if currency is not USD
            if(currency !== "usd"){
              const valueInUSD = convertToUSD(currency, Number(value))
              dealsTotalValueUSD += valueInUSD
            } else {
              //add without conversion if in USD
              dealsTotalValueUSD += Number(value)
            }
        })
        console.log("TOTAL VAL", dealsTotalValueUSD)
        //format number with commas
        dealsTotalValueUSD = dealsTotalValueUSD.toLocaleString("en-US")
        return dealsTotalValueUSD
    }

    const getLocation = (geoIps) => {
        console.log("GE", geoIps)
        let location = ''
        //go through each geoIps ID 
        geoIps.forEach(geoIpsID => {
            //access geoIps object in geoIpsObjects with each geoIps ID
            const geoIpsObj = geoIpsObject[geoIpsID]
            //accesss the geoAddrID property in each object
            const geoAddrID = geoIpsObj["geoaddrid"]
            //access geoAddr object in geoAddressObjects with geoAddr ID
            const geoAddressObject = geoAddressesObject[geoAddrID]
            //get city, state, country
            const country = geoAddressObject["country2"]
            const city = geoAddressObject["city"]
            const stateAbbreviation = getStateAbbreviation(geoAddressObject["state"])
            location = `${city}, ${stateAbbreviation}, ${country}`
            console.log("LOCATION", location)
        })
        return location
    }

    return (
        <div className="contact-rows">
            {contacts.map((contact) => {
                const { firstName, lastName, geoIps, contactTags, deals } = contact
                console.log('___________________________________')
                const name = `${firstName} ${lastName}`
                const initials = `${firstName[0]}${lastName[0]}`
                console.log('NAME', name)
                const contactTagsStr = getContactTags(contactTags)
                const numberOfDeals = deals.length
                const dealsTotalValue = getDealsTotalValue(deals)

                let geoAddressStr = ""
                console.log("GEO", geoIps.length)
                if(geoIps.length === 0){
                    geoAddressStr = 'N/A'
                } else {
                  geoAddressStr = getLocation(geoIps)
                  console.log('GA ARR', geoAddressStr)
                  console.log('___________________________________')
                }

                return (
                  <Contact 
                      name={name}
                      initials={initials}
                      contactTagsStr={contactTagsStr}
                      numberOfDeals={numberOfDeals}
                      dealsTotalValue={dealsTotalValue}
                      geoAddressStr={geoAddressStr}
                  />
                )
            })}
        </div>
    )
}

export default ContactRows