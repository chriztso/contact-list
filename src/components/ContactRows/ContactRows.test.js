import { shallow } from 'enzyme';
import ContactRows from '../ContactRows/ContactRows'
import Contact from '../Contact/Contact'

describe('ContactRows', () => {
    it('renders ContactRows', () => {
        const contacts = []
        const dealsObject = {}
        const tagsObject = {}
        const contactTagsObject = {}
        const geoIpsObject = {}
        const geoAddressesObject = {}
        const wrapper = shallow(
            <ContactRows  
              contacts={contacts}
              dealsObject={dealsObject}
              tagsObject={tagsObject}
              contactTagsObject={contactTagsObject}
              geoIpsObject={geoIpsObject}
              geoAddressesObject={geoAddressesObject}
            />);
        expect(wrapper).toMatchSnapshot();
    })
    
})