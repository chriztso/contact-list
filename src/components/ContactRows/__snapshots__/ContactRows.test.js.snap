// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactRows renders ContactRows 1`] = `
<div
  className="contact-rows"
/>
`;
